File: Upcoder/Lập trình cơ bản/5.Mảng/Tìm max mảng 2 chiều/TimMax2.py
n, m = map(int, input().split())

a = []
print()
for i in range(n):
    row = list(map(int, input().split()))
    a.append(row)

max_value = a[0][0]
for i in range(n):
    for j in range(m):
        if a[i][j] > max_value:
            max_value = a[i][j]

print(max_value, end="")

File: Upcoder/Lập trình cơ bản/5.Mảng/Tìm max mảng 2 chiều/TimMax2.cs
using System;

namespace Mang2c
{
    public class Program
    {
        static void Main(string[] args)
        {
            // Đọc n và m từ đầu vào
            string[] input = Console.ReadLine().Split(' ');
            int n = int.Parse(input[0]);
            int m = int.Parse(input[1]);
            
            int[,] a = new int[n, m];
            
            for (int i = 0; i < n; i++)
            {
                string[] row = Console.ReadLine().Split(' ');
                for (int j = 0; j < m; j++)
                {
                    a[i, j] = int.Parse(row[j]);
                }
            }
            
            int max = a[0, 0];
            for (int i = 0; i < n; i++)
            {
                for (int j = 0; j < m; j++)
                {
                    if (max < a[i, j])
                        max = a[i, j];
                }
            }
            
            Console.Write(max);
        }
    }
}
